src/serialize.rs

```diff
@@ -13,6 +13,12 @@ pub struct SerializeConfig {
     pub root_eclasses: Vec<(ArcSort, Value)>,
 }
 
+struct Serializer<'a> {
+    extractor: Extractor<'a>,
+    node_ids: NodeIDs,
+    result: egraph_serialize::EGraph,
```

Comment:
nit: Is the reason for this struct just to minimize the number of args to pass down to functions? Maybe just stylistic, but I find it a little easier to follow if the extractor was just passed down explicitly as another arg, instead of being collected in this new struct.

Reply:
I'm in favor of having the struct now that the code has reached a sufficient level of complexity

Reply:
Sounds good!